### PR TITLE
feat: Do not use default values in the Kubernetes yamls

### PR DESCRIPTION
**Pull request by Akash-Nayak**
Addresses https://github.com/konveyor/move2kube/issues/940

Old deployment yaml

```console
apiVersion: apps/v1
kind: Deployment
metadata:
  creationTimestamp: null
  labels:
    move2kube.konveyor.io/service: nodejs
  name: nodejs
spec:
  progressDeadlineSeconds: 600
  replicas: 2
  revisionHistoryLimit: 10
  selector:
    matchLabels:
      move2kube.konveyor.io/service: nodejs
  strategy:
    rollingUpdate:
      maxSurge: 25%
      maxUnavailable: 25%
    type: RollingUpdate
  template:
    metadata:
      creationTimestamp: null
      labels:
        move2kube.konveyor.io/service: nodejs
      name: nodejs
    spec:
      containers:
        - image: quay.io/default-myproject/nodejs:latest
          imagePullPolicy: Always
          name: nodejs
          ports:
            - containerPort: 8080
              protocol: TCP
          resources: {}
          terminationMessagePath: /dev/termination-log
          terminationMessagePolicy: File
      dnsPolicy: ClusterFirst
      imagePullSecrets:
        - name: quay-io-imagepullsecret
      restartPolicy: Always
      schedulerName: default-scheduler
      securityContext: {}
      terminationGracePeriodSeconds: 30
status: {}
```

New deployment yaml

```console
apiVersion: apps/v1
kind: Deployment
metadata:
  creationTimestamp: null
  labels:
    move2kube.konveyor.io/service: nodejs
  name: nodejs
spec:
  replicas: 2
  selector:
    matchLabels:
      move2kube.konveyor.io/service: nodejs
  strategy: {}
  template:
    metadata:
      creationTimestamp: null
      labels:
        move2kube.konveyor.io/service: nodejs
      name: nodejs
    spec:
      containers:
        - image: quay.io/no-def-new/nodejs:latest
          imagePullPolicy: Always
          name: nodejs
          ports:
            - containerPort: 8080
          resources: {}
      imagePullSecrets:
        - name: quay-io-imagepullsecret
      restartPolicy: Always
      securityContext: {}
status: {}
```

I have tested enterprise-app deployment on the OCP cluster with the new `deploy/yamls` files and it's working fine.

Signed-off-by: Akash Nayak <akash19nayak@gmail.com>

### CODE REVIEW

File: assets/built-in/transformers/kubernetes/argocd/transformer.yaml [hunk 1]
```diff
@@ -17,3 +17,6 @@ spec:
   produces:
     KubernetesYamls:
       disabled: false
+  config:
+    outputPath: "deploy/cicd/argocd"
+    useDefaultValuesInYamls: false
```
Comment (ashokponkumar) [inline assets/built-in/transformers/kubernetes/argocd/transformer.yaml:22]: would `setDefaultValuesInYamls` be a better name?

File: transformer/kubernetes/kubernetestransformer.go [hunk 2]
```diff
@@ -46,8 +47,9 @@ type Kubernetes struct {
 
 // KubernetesYamlConfig stores the k8s related information
 type KubernetesYamlConfig struct {
-	IngressName string `yaml:"ingressName"`
-	OutputPath  string `yaml:"outputPath"`
+	IngressName                string `yaml:"ingressName"`
+	OutputPath                 string `yaml:"outputPath"`
+	UseDefaultValuesInK8sYamls bool   `yaml:"useDefaultValuesInK8sYamls"`
```
Comment (ashokponkumar) [inline transformer/kubernetes/kubernetestransformer.go:52]: Would it be better to use the same name in all configs? like say `setDefaultValuesInYamls`?

**Review by ashokponkumar — CHANGES_REQUESTED** [review]
Really nice work Akash! I like it! A few suggestions..

File: assets/built-in/transformers/kubernetes/argocd/transformer.yaml (hunk 1 above)
Comment (Akash-Nayak) [reply assets/built-in/transformers/kubernetes/argocd/transformer.yaml:22]: Yes sure

File: transformer/kubernetes/kubernetestransformer.go (hunk 2 above)
Comment (Akash-Nayak) [reply transformer/kubernetes/kubernetestransformer.go:52]: Yes sure

File: assets/built-in/transformers/kubernetes/argocd/transformer.yaml (hunk 1 above)
Comment (Akash-Nayak) [reply assets/built-in/transformers/kubernetes/argocd/transformer.yaml]: Done.

File: transformer/kubernetes/kubernetestransformer.go (hunk 2 above)
Comment (Akash-Nayak) [reply transformer/kubernetes/kubernetestransformer.go]: Done.

**Comment by Akash-Nayak** [comment]
> Really nice work Akash! I like it! A few suggestions..

Thank you very much, Ashok ji! :-)

**Review by ashokponkumar — CHANGES_REQUESTED** [review]
Might be good to rename the internal variables to be set too.

**Comment by Akash-Nayak** [comment]
> Might be good to rename the internal variables to be set too.

Yes sure, done. I missed renaming the internal variables while I was doing case-sensitive find and replace.

**Pull request merged**